src/tools/kotlin-entity-generator.ts

```diff
@@ -302,12 +233,15 @@ ${expression}
                 val entityId = decoder.decodeText()
                 decoder.validate("|")
                 ${fieldCount > 0 ? (`
-                ${ktUtils.indentFollowing(this.setFieldsToDefaults, 3)}
+                ${ktUtils.indentFollowing(this.mapFields(({escaped, defaultVal}) => `var ${escaped} = ${defaultVal}`), 3)}
```

Comment:
I like with the `mapFields` function, the field-associated code block can be defined in context with the rest of the module.